releasenotes/notes/add-passmanager-module-3ae30cff52cb83f1.yaml

```diff
@@ -3,24 +3,93 @@ features:
   - |
     A new module :mod:`qiskit.passmanager` is added.
     This module implements a generic pass manager and flow controllers,
-    and provides an infrastructure to manage execution of transform passes.
+    and provides an infrastructure to manage execution of pass manager tasks.
     The pass manager is a baseclass and not aware of the input and output object types,
-    and you need to create a subclass of the pass manager
-    for a particular program data to optimize.
+    and subclass must be created for a particular program type to optimize.
     The :mod:`qiskit.transpiler` module is also reorganized to rebuild the existing
-    quantum circuit pass manager based off of new generic pass manager.
+    pass manager based off of the generic pass manager.
     See upgrade notes for more details.
 upgrade:
   - |
-    :class:`qiskit.transpiler.PassManager` is now a subclass of
-    :class:`qiskit.passmanager.BasePassManager`. There is no functional modification
-    due to this class hierarchy change.
+    New base class for passes :class:`.GenericPass` and flow controllers :class:`.BaseController`
+    are introduced in the :mod:`qiskit.passmanager` module.
+    Because the flow controller is a collection of passes and a controller can be recursively
+    nested into the task pipeline, new classes are designed with the idea of
+    the composite pattern, and the interface class :class:`qiskit.passmanager.base_task.Task`
+    is also introduced. This class defines a signature of :meth:`Task.execute` method.
+    This unified design eliminates complexity of the conventional pass manager;
+    the execution logic dispatch and task structure renormalization
+    are no longer necessary, whom the :class:`RunningPassManager` used to be responsible for.
+    Existing flow controllers :class:`.FlowControllerLinear`,
+    :class:`.ConditionalController`, and :class:`.DoWhileController` are now subclass of
+    the :class:`.BaseController`.
+    Note that these controllers are no longer iterable, as they drop the implementation of
+    :class:`__iter__` method.
```

Comment:
I was also thinking about backward compatibility. Even if we implement `__iter__` for backward compatibility, it will be never practically called because new `execute` method doesn't assume a controller being iterator. I think communicating through the release note is most effective.